### PR TITLE
[flutter_tools] Add violating plugin name to validation errors

### PR DESCRIPTION
**Pull request by MrAlek**
## Description

Adds the name of the violating plugin to error messages when validating plugin specifications in `flutter packages get`, etc so it's easier to track down which is the violating plugin.

Console output from `flutter packages get` before:

```
Invalid plugin specification.
```

After:

```
Invalid plugin specification test_plugin_name.
```

## Tests

I added the following tests:

- `Legacy Format and Multi-Platform Format together is not allowed and error message contains plugin name` in `plugin_parsing_test.dart`
## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [X] No, no existing tests failed, so this is _not_ a breaking change.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/plugin_parsing_test.dart [hunk 1]
```diff
@@ -122,6 +123,17 @@ void main() {
...
 
+    test('Legacy Format and Multi-Platform Format together is not allowed and error message contains plugin name', () {
+      const String pluginYamlRaw = 'androidPackage: com.flutter.dev\n'
+          'platforms:\n'
+          ' android:\n'
+          '  package: com.flutter.dev\n';
+
+      final YamlMap pluginYaml = loadYaml(pluginYamlRaw) as YamlMap;
+      final Plugin Function() parsePlugin = () => Plugin.fromYaml(_kTestPluginName, _kTestPluginPath, pluginYaml, const <String>[]);
+      expect(parsePlugin, throwsA(predicate<ToolExit>((ToolExit e) => e.message.contains(_kTestPluginName))));
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/plugin_parsing_test.dart]: I think we would normally write this as:
```
expect(
  () => Plugin.fromYaml(_kTestPluginName, _kTestPluginPath, pluginYaml, const <String>[]),
  throwsToolExit(message: _kTestPluginName),
);
```

**Review by zanderso — APPROVED** [review]
lgtm w/ nit

**Comment by MrAlek** [comment]
@zanderso Fixed nit

File: packages/flutter_tools/test/general.shard/plugin_parsing_test.dart [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:flutter_tools/src/base/common.dart';
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/plugin_parsing_test.dart]: The presubmit analyzer pass is saying that this import is unused and can be removed.

Comment (MrAlek) [reply packages/flutter_tools/test/general.shard/plugin_parsing_test.dart]: That's right, it was used before fixing the nit. Removed.

**Pull request merged**